app/models/forms/export.rb

```diff
@@ -149,8 +154,20 @@ def to_xls
             # add "and" unless we're at the end
             constraints_to_push += " and " unless c_index + 1 == q.constraints.length
 
-            # TODO: add support for constraint messages ("rejection_msg" in NEMO)
+            # Write translated constraint message columns ("rejection_msg" in NEMO)
             # https://xlsform.org/en/#constraint-message
+            # NEMO allows multiple constraint messages for each rule, whereas XLSForm only supports one message per row.
+            # Thus, if there are multiple constraints or rules for this question, combine all provided messages into one string (per locale), separated by a semicolon
+            locales.each_with_index do |locale, locale_index|
+              # Attempt to get a message for that constraint for that language (may be nil if a translation is not provided)
+              constraint_message = c.rejection_msg_translations&.dig(locale.to_s)
+
+              if constraint_message.present?
+                constraint_msg_to_push[locale_index] += constraint_message
+                # Add semicolon concatenator, unless at the end
+                constraint_msg_to_push[locale_index] += "; " unless c_index + 1 == q.constraints.length
```

Comment:
this works perfectly fine, just sharing a helpful alternative: `[1, 2, 3].join("; ")`